integr8ly/grafana-operator
# fix datasource json and secure json fields

Fixes #38 

The problem was that grafana uses a type of `map[string]interface{}` for the jsonData and secureJsonData fields of a data source. The operator-sdk does not support code generation for `interface{}` fields. Instead of only allowing strings we're now providing structs for the most common jsonData and secureJsonData options as documented [here](https://grafana.com/docs/administration/provisioning/#datasources).

Verification steps:

1. Install the operator using the `quay.io/integreatly/grafana-operator:fix-extra-json` tag.
2. Create the example data source.
3. The operator will restart grafana.
4. After a few seconds, log in to Grafana and check the data sources.
5. The Prometheus data source should exist and the `tlsSkipVerify` option should be checked.
